app/models/post.rb

```diff
@@ -1,5 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
   has_many :likes, dependent: :destroy
   has_many :comments, dependent: :destroy
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] [OPTIONAL] Please consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database, then make the `#update_posts_counter` method private. (you can do the following to the rest of the `update` methods in other models)

Reply:
Thanks for your review and suggestion. I will consider your suggestion.